### PR TITLE
Fix bug where EXISTS did not accept maps

### PR DESCRIPTION
**Pull request by Mats-SX**

### CODE REVIEW

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Predicate.scala [hunk 1]
```diff
@@ -152,6 +152,7 @@ case class PropertyExists(identifier: Expression, propertyKey: KeyToken) extends
   def isMatch(m: ExecutionContext)(implicit state: QueryState): Option[Boolean] = identifier(m) match {
     case pc: Node         => Some(propertyKey.getOptId(state.query).exists(state.query.nodeOps.hasProperty(pc.getId, _)))
     case pc: Relationship => Some(propertyKey.getOptId(state.query).exists(state.query.relationshipOps.hasProperty(pc.getId, _)))
+    case IsMap(map)       => Some(map(state.query).contains(propertyKey.name))
```
Comment (davidegrohmann) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Predicate.scala]: I think you can use `IsMap` also for node and relationships.

File: community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/LoadCsvAcceptanceTest.scala [hunk 2]
```diff
@@ -123,6 +123,28 @@ class LoadCsvAcceptanceTest
...
+      writer =>
+        writer.println("id,name,x")
+        writer.println("1,'Aardvark',0")
+        writer.println("2,'Babs'")
+        writer.println("3,'Cash',1")
+        writer.println("4,'Dice',\"\"")
+        writer.println("5,'Emerald',")
+    })
+    for (url <- urls) {
+      val result = execute(s"LOAD CSV WITH HEADERS FROM '$url' AS line WITH line WHERE line.name IS NOT NULL RETURN line.x")
```
Comment (davidegrohmann) [inline community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/LoadCsvAcceptanceTest.scala]: maybe `LOAD CSV ... line.x IS NOT NULL RETURN line.name` is a more meaningful test to have

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Predicate.scala (hunk 1 above)
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Predicate.scala]: but in this case you want to treat nodes and relationships separately, e.g. do `state.query.nodeOps.hasProperty(pc.getId, _)` for nodes

File: community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/LoadCsvAcceptanceTest.scala (hunk 2 above)
Comment (Mats-SX) [inline community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/LoadCsvAcceptanceTest.scala]: Yes, indeed. And this exposed us to actually not have implemented the correct fix... Good catch!

**Pull request merged**